stan/math/rev/mat/fun/mdivide_left_tri.hpp

```diff
@@ -57,25 +57,54 @@ class mdivide_left_tri_vv_vari : public vari {
           variRefA_[pos++] = A(i, j).vi_;
     }
 
-    Map<matrix_d> a_map(A_, M_, M_);
     Map<matrix_d> c_map(C_, M_, N_);
+    Map<matrix_d> a_map(A_, M_, M_);
     a_map = A.val();
     c_map = B.val();
     Map<matrix_vi>(variRefB_, M_, N_) = B.vi();
-    c_map = a_map.template triangularView<TriView>().solve(c_map);
+#ifdef STAN_OPENCL
+    if (A.rows()
+        >= opencl_context.tuning_opts().tri_inverse_size_worth_transfer) {
+      matrix_cl<double> A_cl(a_map, from_eigen_uplo_type(TriView));
+      matrix_cl<double> C_cl(c_map);
+      C_cl = tri_inverse(A_cl) * C_cl;
+      c_map = from_matrix_cl(C_cl);
+    } else {
+#endif
+      c_map = a_map.template triangularView<TriView>().solve(c_map);
+#ifdef STAN_OPENCL
+    }
+#endif
+
     Map<matrix_vi>(variRefC_, M_, N_)
         = c_map.unaryExpr([](double x) { return new vari(x, false); });
   }
 
   virtual void chain() {
     using Eigen::Map;
-
-    matrix_d adjB = Map<matrix_d>(A_, M_, M_)
-                        .template triangularView<TriView>()
-                        .transpose()
-                        .solve(Map<matrix_vi>(variRefC_, M_, N_).adj());
-    matrix_d adjA = -adjB * Map<matrix_d>(C_, M_, N_).transpose();
-
+    matrix_d adjA;
+    matrix_d adjB;
+
+#ifdef STAN_OPENCL
+    if (M_ >= opencl_context.tuning_opts().tri_inverse_size_worth_transfer) {
+      matrix_d tmp_variRefC = Map<matrix_vi>(variRefC_, M_, N_).adj();
+      matrix_cl<double> A_cl(A_, M_, M_, from_eigen_uplo_type(TriView));
+      matrix_cl<double> C_cl(C_, M_, N_);
+      matrix_cl<double> variRefC_cl(tmp_variRefC);
```

Comment:
We can get rid of the `tmp_variRefC` above
```suggestion
      matrix_cl<double> variRefC_cl(Map<matrix_vi>(variRefC_, M_, N_).adj());
```